resource-manager/src/main/java/org/ihtsdo/otf/resourcemanager/ResourceManager.java

```diff
@@ -1,113 +1,409 @@
+        final String fullPath = getFullPath(resourcePath);
+        try {
+            return resourceLoader.getResource(fullPath).getInputStream();
+        } catch (AmazonS3Exception e) {
+            throw new IOException("Failed to load resource '" + fullPath + "'.", e);
+        }
+    }
+
+    /**
+     * Reads the resource from the given <code>resourcePath</code>. It will
+     * return {@code null} if the {@code Resource} does not exist.
+     *
+     * @param resourcePath Path to the resource.
+     * @return an {@code InputStream} which corresponds to the resource stream.
+     * Otherwise, it will return {@code null} if the {@code Resource} does not
+     * exist.
+     * @throws IOException If an error occurs while trying to load the resource.
+     */
+    public InputStream readResourceStreamOrNullIfNotExists(final String resourcePath) throws IOException {
+        try {
+            final Resource resource = resourceLoader.getResource(getFullPath(resourcePath));
+            return resource.exists() ? resource.getInputStream() : null;
+        } catch (AmazonS3Exception e) {
+            throw new IOException("Failed to load resource '" + resourcePath + "'.", e);
+        }
+    }
+
+    /**
+     * Writes the data to the given resource path location.
+     *
+     * @param resourcePath        Which points to the resource that is
+     *                            going to have the write operation performed
+     *                            on it.
+     * @param resourceInputStream Input stream which represents the converted
+     *                            object, encoded as bytes.
+     * @throws IOException If an error occurs while trying to write the
+     *                     input stream to the specified resource path.
+     */
+    public void writeResource(final String resourcePath,
+                              final InputStream resourceInputStream) throws IOException {
+        try {
+            try (final OutputStream outputStream = writeResourceStream(resourcePath);
+                 final InputStream inputStream = resourceInputStream) {
+                StreamUtils.copy(inputStream, outputStream);
+            }
+        } catch (AmazonS3Exception e) {
+            throw new IOException("Failed to write resource '" + resourcePath + "'.", e);
+        }
+    }
+
+    /**
+     * Writes the data to the given resource path location and returns the
+     * {@code OutputStream}.
+     *
+     * @param resourcePath Which points to the resource that is going
+     *                     to have the write operation performed on it.
+     * @return an {@code OutputStream} which is the result after getting
+     * the resource from the given path.
+     * @throws IOException If an error occurs while trying to get the
+     *                     {@code OutputStream}.
+     */
+    public OutputStream writeResourceStream(final String resourcePath) throws IOException {
```

Comment:
How does this method handle writing to the cloud?
In general we try to make the ResourceManager write to disk or cloud transparently via configuration changes. The idea is that an application using this class could switch between using disk or cloud storage via a configuration change and without any code changes. For that to work most methods need to check the config and be able to deal with both storage types.

Reply:
This existed before my changes were added but I believe this method should be private & name changes to reflect what it actually does as it's used as part of the other writing wrapper

Reply:
It seems like this method is used for just reading the resource from cloud and then returns the value for the other write method to have the output stream so that it can write the new input stream into.

Reply:
I will look into enhancing this method so that it makes more sense/readable but I don't think it should be public

Reply:
Fixed.